### ISSUE TITLE
events.php, tasks.php and strategies.php ignored in .rocketeer/stages/[stagename]/

### ISSUE THREAD

**Issue by petermelis** [rocketeers/rocketeer#599]
When I use the file-based alternative for setting up different configs for different stages, it works beautifully with for example hooks.php

hooks.php is located in .rocketeer/stages/staging/ This config is used when deploying to stage="staging".

If I add an events.php, or strategies.php to the .rocketeer/stages/staging/ folder, they are ignored. The config in these files is not used when deploying to stage="staging"

Can someone tell me why and if and how it can be fixed

Thanks!

**Comment by Anahkiasen** [comment]
This is normal behavior, cascading only works for config files. That being said it should work for `strategies.php` are you sure it's not loaded?

**Comment by petermelis** [comment]
Yes, when I alter the file in .rocketeer to only use composer. It only uses composer. When I alter the file in  .rocketeer/stages/staging/ it is ignored.

**Comment by Anahkiasen** [comment]
Hm maybe this is related to how strategies are bound as singletons to the container, maybe it's not overriding them

**Comment by petermelis** [comment]
It certainly looks like that. Is that fixable?

*Peter Melis*
Product Manager

*T* +31 (0)6 33 200 200
*E* peter.melis@britelayer.com
*I* www.britelayer.com

NEW ADDRESS:
Koningin Wilhelminaweg 469
3737BE Groenekan, The Netherlands



2015-09-10 11:23 GMT+02:00 Maxime Fabre <notifications@github.com>:

**Comment by Anahkiasen** [comment]
I think it already is on 3.0 because IIRC strategies are not bound to the container anymore but created on the fly, but on 2.0 it'd require some changes so need to see.

**Comment by petermelis** [comment]
Thank you for your quick responses!

I would appreciate it if you would keep me up-to-date on this.

Is 3.0 still being developed or in beta?

*Peter Melis*
Product Manager

*T* +31 (0)6 33 200 200
*E* peter.melis@britelayer.com
*I* www.britelayer.com

NEW ADDRESS:
Koningin Wilhelminaweg 469
3737BE Groenekan, The Netherlands



2015-09-10 11:36 GMT+02:00 Maxime Fabre <notifications@github.com>:

**Comment by Anahkiasen** [comment]
It's still being developed in the `develop` branch

**Comment by petermelis** [comment]
Ok, thanks.

Is it in any other way possible to use another deploy strategy (sync
instead of clone) on another stage?


*Peter Melis*
Product Manager

*T* +31 (0)6 33 200 200
*E* peter.melis@britelayer.com
*I* www.britelayer.com

NEW ADDRESS:
Koningin Wilhelminaweg 469
3737BE Groenekan, The Netherlands



2015-09-10 12:00 GMT+02:00 Maxime Fabre <notifications@github.com>:
